pkg/fipcontroller/controller.go

```diff
@@ -41,77 +42,87 @@ func NewController(config *Configuration) (*Controller, error) {
 		HetznerClient:    hetznerClient,
 		KubernetesClient: kubernetesClient,
 		Configuration:    config,
-		NodeName:         os.Getenv("NODE_NAME"),
 	}, nil
 }
 
 func NewControllerConfiguration() (*Configuration, error) {
 	var config Configuration
 
+	// Read config from file if present
 	file, err := ioutil.ReadFile("config/config.json")
-	if err != nil {
-		return nil, fmt.Errorf("could not open config file: %v", err)
+	if err == nil {
```

Comment:
This should be `if err != nil {`. That way you can early return if you're unable to read the file or unable to unmarshal.

```go

	file, err := ioutil.ReadFile("config/config.json")
	if err != nil {
		return nil, fmt.Errorf("failed to read config: %v", err)
	}
	if err := json.Unmarshal(file, &config); err != nil {
		return nil, fmt.Errorf("could not decode config: %v", err)
	}
```

Reply:
The basic idea was that the config file is optional. Rewrote a little :)